src/serialize.h

```diff
@@ -203,6 +203,7 @@ template<typename Stream> inline void Serialize(Stream& s, int64_t a ) { ser_wri
 template<typename Stream> inline void Serialize(Stream& s, uint64_t a) { ser_writedata64(s, a); }
 template<typename Stream, int N> inline void Serialize(Stream& s, const char (&a)[N]) { s.write(MakeByteSpan(a)); }
 template<typename Stream, int N> inline void Serialize(Stream& s, const unsigned char (&a)[N]) { s.write(MakeByteSpan(a)); }
+template <typename Stream, typename B, std::size_t N> void Serialize(Stream& s, const std::array<B, N>& a) { (void)/* force byte-type */UCharCast(a.data()); s.write(MakeByteSpan(a)); }
```

Comment:
nit: ser, that's a long line

Reply:
Seemed awkward to format this line, but leave the others. I think I'll leave this as is.

Reply:
Might be clearer as `{ s.write(AsBytes(UCharSpanCast(Span(a)))); }` ? I think that has all the same restrictions as you get with the dummy `UCharCast`.

Reply:
Yeah. Might be another place where C++20 concepts would be good -- `concept BytesLike = requires (B& b) { UCharCast(b) }` and `template<typename Stream, BytesLike B, ..>` or so.